### PR TITLE
Sites: Display button for resuming last edited draft

### PR DESCRIPTION
**Pull request by aduth**
This pull request seeks to introduce a new button to the master bar for resuming a draft that had been edited during the current page session.

New Post

Detail:
image

__Testing instructions:__

Verify that a new "Continue editing" button is shown in the master bar for the last edited post, only for the duration of your current page session, and is removed after the post is published.

/cc @mtias

### CODE REVIEW

File: client/components/site-icon/style.scss [hunk 1]
```diff
@@ -9,13 +5,13 @@
...
+	text-align: center;
 
 	// Globe icon for sites without an icon
 	&.is-blank {
-		.noticon {
+		.gridicon {
 			color: $white;
-			text-align: center;
-			z-index: z-index( 'root', '.site-icon.is-blank .noticon' );
+			z-index: z-index( 'root', '.site-icon.is-blank .gridicon' );
```
Comment (aduth) [inline client/components/site-icon/style.scss]: We need to update this in the [z-index mapping](https://github.com/Automattic/wp-calypso/blob/1d0480ea45da81f3c06f46abf4844f0cb080058d/assets/stylesheets/shared/functions/_z-index.scss#L38).

Comment (gwwar) [inline client/components/site-icon/style.scss]: You may find https://github.com/gwwar/z-context useful, though the extension is still a bit rough.

**Comment by ryanboren** [comment]
If you enter the editor for a draft and exit without changes, the button doesn't update. An actual edit is required to update the button. Seems fine and edge.

I tested on an iPhone 6+ verifying that the resume button doesn't show or cause collateral damage. Looks good.

Resume shows on an iPad in both landscape and portrait. 

img_1279
img_1280

I noticed on my iPad Air that the masterbar no longer pins to the top when scrolling down. It disappears offscreen instead of sticking. I see this only when testing the resume branch and not in production. This could be my testing environment. I ran a distclean but maybe something lingers. Can anyone reproduce this?

**Comment by folletto** [comment]
I'm not sure the masterbar placement for that button is something we should pursue. It breaks the consistency of the masterbar across platform (and that's the only thing that stays consistent) and we know how hard it is to keep that consistency in place.

Plus, won't work on mobile — thus a solution helpful there would be ideal.

**Comment by apeatling** [comment]
+1 -- I like this idea, but the masterbar placement is not ideal for the reasons @folletto mentions.

**Comment by folletto** [comment]
_Aside: should we make the Draft page as default on the "Blog Posts" landing? That would make this one click shorter already from that route. :) _

**Comment by aduth** [comment]
>It breaks the consistency of the masterbar across platform

I could appreciate this argument for permanent fixtures of the master bar. However, in this case the action item is very obviously transient; it's discarded between sessions, and is only active after navigating away from editor. Arbitrarily enforcing some hard-and-fast rule against revisions to the master bar seems short-sighted. What exactly are the costs and benefits we're weighing here? As it stands, accidentally navigating away from the editor is a distressing experience, and requires a series of well-executed steps to resume from where a user had left off. I'd argue the changes here introduce a simple and obvious solution to the problem. The downsides are... it's not present in mobile?

Note also that this isn't enabled in production. At the very least, it seems worthwhile to conduct some user testing before discarding it on an ideological basis.

I'll let @mtias weigh in on this more, as he's the one who roped me into this hack day project :wink: 

I'm curious in exploring other ideas in solving the original problem though. Some other ideas that had been discussed include:
- Adding an option to the editor button site selector
 - Most users are single-site users, and editor button navigates them directly to editor. This would be a new (and potentially disruptive) step.
 - We tried a similar idea before with choosing between starting a new post and a new page, and it provied to be a convoluted experience in testing
- Via a modal or other prompt, request that the user to choose whether to start a new draft or from one of their recent drafts when landing at the editor
 - Not as discoverable (need to know that navigating to editor would prompt them)
 - Disruptive to workflows where user expects editor button to start new draft
- Improve discoverability of drafts drawer
 - Similar discoverability concerns. Particularly, these solutions don't address the state of confusion after leaving the editor and wanting to immediately return to where they had left off.
- Add a "Drafts" button next to "New" in My Sites sidebar
 - Similar idea to @folletto on defaulting to drafts in Blog Posts, at least in reducing number of steps to return to editor

Refs: p58i-3xf-p2, pb6Nl-aAl-p2

**Comment by folletto** [comment]
> Arbitrarily enforcing some hard-and-fast rule against revisions to the master bar seems short-sighted.

It's not arbitrary, nor short sighted, nor ideological. It's an outcome of a long process that took years of work and coordination across teams. It's exactly because we have a long term view that we need to be more strict on that specific component. We came out rather recently from a situation where every single screen we had had a different bar, every decision that touches it needs to be done twice as carefully, and across all the screens.

That said, the masterbar _can_ be touched, with care, but the rule of thumb is to avoid it. 

> I'm curious in exploring other ideas in solving the original problem though. 

I'd be totally up for that. Any solution that works destkop-mobile-iOS-Android I think it's a good way to go. :)

1. "Adding an option to the editor button site selector" seems sensible to me. Previous attempts were done in a different context, so we could probably find a way that works. I also imagine maybe an orange dot on the icon itself to signal the extra option. And we need a different design then to handle single sites, yes.
2. "Via a modal or other prompt" agree on the assessment of this one. We could maybe do something slightly different as hinting that there's a "new draft"? So even if you get in to create a new one you see it? But I agree on the discoverability bit. This is hard.
3. "Improve discoverability of drafts drawer", as I suggested above I'd try to make more evident in our main navigation the Drafts themselves, because it would shorten the taps to the editor. An orange dot on a "leftover" draft might then be enough to signal it tho.
4. Two buttons might crowd it too much, but maybe we can do a design exploration and see what happens.

All the above should work well on all our platforms, so :+1:  from that perspective.

**Comment by folletto** [comment]
> or start using the main content real state with floating items like notices do.

This could be another good idea to explore. Using the pill with the message and also a dismiss button (so I can just close it) could be a nice approach, and compatible everywhere as it's reusing an existing design pattern for temporary messages. :)

**Comment by aduth** [comment]
>>Arbitrarily enforcing some hard-and-fast rule against revisions to the master bar seems short-sighted.
>
>It's not arbitrary, nor short sighted, nor ideological. It's an outcome of a long process that took years of work and coordination across teams.

I should be clear that I never meant to imply that the cautiousness by which we approach changes to the masterbar was arbitrarily decided upon, but that the immediate dismissal for a proposed changes on the grounds that it affects said area, without proper discussion of its merits, is an _arbitrary enforcement_ of that cautiousness.

That being said, I'm glad we've evolved this discussion into more of an exploration of the options available to us, and I'm sorry if I came across terse in my earlier message.

**Comment by mtias** [comment]
> Using the pill with the message and also a dismiss button (so I can just close it) could be a nice approach

I don't think this is a great use of the notice pattern for something that is purely a navigational item. It also partially obscures content and encourages taking action quickly.

**Comment by folletto** [comment]
> I should be clear that I never meant to imply that the cautiousness by which we approach changes to the masterbar was arbitrarily decided upon, but that the immediate dismissal of a proposed change on the grounds that it affects said area, without proper discussion of its merits, is an arbitrary enforcement of that cautiousness.

Sorry. I didn't want to sound as a dismissal, but as a reference point that I felt hasn't been discussed before making the proposal above.

> I don't think this is a great use of the notice pattern for something that is purely a navigational item. It also partially obscures content and encourages taking action quickly.

...which, to me, is the point? Why would I want to keep that thing visible for a whole session, if I didn't want to go back to it quickly? :)

**Comment by mtias** [comment]
I think that's when we start assuming too much :) I may be reading something in the reader and don't want the distraction of a notice, I may be switching through a couple sections to check something up, etc. It may be cool to test the two approaches, though, if those are the best candidates we have.

**Comment by folletto** [comment]
:+1:

**Comment by folletto** [comment]
Quick mock just to keep che conversation going:

draft-as-notice-custom-i2

It would be perfect if attached to that there was a logic like:

_If you dismiss this box 5 times always on the next page after navigating away, we'll ask if you want to keep the functionality, if not, we'll tell you how to access your drafts._

**Comment by timmyc** [comment]
Wanted to give this PR a bump since there was lively discussion here, but no real clear course of action regarding a next step.  I think the idea has great value in increasing engagement and is very worth exploring more.  I see in its current state, this feature is only enabled on `wpcalypso` and `development`... and I also do understand the amount of work that has gone into the masterbar standard thus far.  Would it be okay to consider just deploying this to limited environments to test it out a bit - albeit internally or maybe on horizon?

One other possible placement could be in the site selector that is shown when clicking the new post icon.  We could possibly show the latest draft under each site and allow a quick shortcut to the prior draft there too.

**Comment by folletto** [comment]
I'd still prefer something that could be dismissed like the bubble above, however, I'd be cool in testing this on Horizon to see what a larger pool of users will say. :)

**Comment by aduth** [comment]
Pushed a final rebase, including feature flag enabling for Horizon. @mtias had suggested we enable the feature in desktop, but since that's a "live" environment, I'd rather let this simmer a couple days in testing environments to ensure any bugs have been shaken loose.

Planning to merge shortly.

**Pull request merged**